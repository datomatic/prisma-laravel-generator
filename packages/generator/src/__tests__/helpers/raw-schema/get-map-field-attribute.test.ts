import {getRawSample} from '../../__fixtures__/get-raw-sample';
import {getSampleDmmf} from '../../__fixtures__/get-sample-dmmf';
import getMapFieldAttribute from '../../../helpers/raw-schema/get-map-field-attribute';

test('getMapFieldAnnotation', async () => {
  const raw = await getRawSample();
  const sampleDMMF = await getSampleDmmf();

  for (const modelInfo of sampleDMMF.datamodel.models) {
    for (const fieldInfo of modelInfo.fields) {
      expect(
        getMapFieldAttribute(fieldInfo.name, modelInfo.name, raw),
      ).toMatchSnapshot(`${modelInfo.name} - ${fieldInfo.name}`);
    }
  }
});
