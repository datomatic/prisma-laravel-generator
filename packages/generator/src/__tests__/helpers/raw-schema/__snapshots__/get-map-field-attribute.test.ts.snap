// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMapFieldAnnotation: Category - id 1`] = `false`;

exports[`getMapFieldAnnotation: Category - name 1`] = `false`;

exports[`getMapFieldAnnotation: Category - posts 1`] = `false`;

exports[`getMapFieldAnnotation: Post - author 1`] = `false`;

exports[`getMapFieldAnnotation: Post - authorId 1`] = `false`;

exports[`getMapFieldAnnotation: Post - categories 1`] = `false`;

exports[`getMapFieldAnnotation: Post - createdAt 1`] = `false`;

exports[`getMapFieldAnnotation: Post - id 1`] = `false`;

exports[`getMapFieldAnnotation: Post - published 1`] = `false`;

exports[`getMapFieldAnnotation: Post - title 1`] = `false`;

exports[`getMapFieldAnnotation: Profile - bio 1`] = `false`;

exports[`getMapFieldAnnotation: Profile - id 1`] = `false`;

exports[`getMapFieldAnnotation: Profile - user 1`] = `false`;

exports[`getMapFieldAnnotation: Profile - userId 1`] = `false`;

exports[`getMapFieldAnnotation: User - email 1`] = `false`;

exports[`getMapFieldAnnotation: User - id 1`] = `false`;

exports[`getMapFieldAnnotation: User - name 1`] = `false`;

exports[`getMapFieldAnnotation: User - posts 1`] = `false`;

exports[`getMapFieldAnnotation: User - profile 1`] = `false`;

exports[`getMapFieldAnnotation: User - role 1`] = `false`;

exports[`getMapFieldAnnotation: UserWithMap - email 1`] = `"user_email"`;

exports[`getMapFieldAnnotation: UserWithMap - id 1`] = `"user_id"`;

exports[`getMapFieldAnnotation: UserWithMap - name 1`] = `"user_name"`;

exports[`getMapFieldAnnotation: UserWithPrimaryKey - email 1`] = `false`;

exports[`getMapFieldAnnotation: UserWithPrimaryKey - id 1`] = `false`;

exports[`getMapFieldAnnotation: UserWithPrimaryKey - name 1`] = `false`;
